tutorials/03_rendering_plugins.md

```diff
@@ -0,0 +1,39 @@
+\page renderingplugin Understanding the Rendering Plugin
+
+This is an introduction to different rendering engines and how they are integrated into the Ignition Physics library.
+
+## Ignition Rendering
+
+The \ref ignition::rendering "Ignition Rendering" library integrates external rendering engines into the Ignition Simulation eco-system.
+It allows users to select from multiple supported rendering engines based on their simulation needs.
+Its plugin interface loads rendering engines at runtime.
+It is also possible to integrate your own selected rendering engine by writing a compatible plugin interface.
+
+#### How to Write Your Own Rendering Plugin
+
+A mocked example of a custom rendering engine plugin can be found [here](https://github.com/ignitionrobotics/ign-rendering/tree/ign-rendering3/examples/hello_world_plugin).  In order
+to make your own custom rendering engine, this example is a good starting point.  There are a few key things which will need to be done in order for a custom rendering engine to function:
+
+ * A singleton Render Engine class which implements the pure virtual functions in [`ignition::rendering::BaseRenderEngine`](https://github.com/ignitionrobotics/ign-rendering/blob/main/include/ignition/rendering/base/BaseRenderEngine.hh).
+ * A plugin class which implements the pure virtual functions in [`ignition::rendering::RenderEnginePlugin`](https://github.com/ignitionrobotics/ign-rendering/blob/main/include/ignition/rendering/RenderEnginePlugin.hh)
+ * Registering the plugin, this line can be seen at the bottom of the [`HelloWorldPlugin`](https://github.com/ignitionrobotics/ign-rendering/tree/ign-rendering3/examples/hello_world_plugin/HelloWorldPlugin.cc) example
+
+Finally, for your custom rendering engine to actually have any functionality, you will need to implement your own `Scene` class, which inherits from and implements the pure virtual functions of [`ignition::rendering::Scene`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/include/ignition/rendering/Scene.hh).  The mocked example simply returns `nullptr` for its `CreateSceneImpl(...)` function, so it may be useful to look at the current `Scene` implementations for the other rendering engines within `ignition::rendering` such as [`ogre`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/ogre/src/OgreScene.cc) or [`optix`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/optix/src/OptixScene.cc).
+
+#### Building and Running Your Rendering Plugin
```

Comment:
can you append ` with Ignition Gazebo` to this heading? 

I think there are potentially others who are interested in building their own render engine plugin without requiring ignition gazebo to run it. That's doable and probably deserves its own example and tutorial.

Reply:
86be9fc553ccaa6a7994609c581be9d3e3b7e649